gcp/rakefiles/secrets.rb

```diff
@@ -3,37 +3,74 @@
 
 class Secrets
 
-  def self.set_secrets()
-    saved_secrets_file_path = "../#{ENV['ENV']}/secrets/#{ENV["TF_VAR_project_id"]}-secrets.yml"
-
-    begin
-      secrets = YAML.load(File.read(saved_secrets_file_path))
-    rescue Errno::ENOENT
-      generate_file = true
-      secrets = Hash.new
-    end
+  KMS_KEYRING = "keyring"
+  KMS_DEFAULT_KEY = "default"
 
-    [ \
-      'couchdb_admin_username', \
-      'couchdb_admin_password', \
-      'couchdb_secret', \
-    ].each do |secret|
-      unless ENV[secret.upcase].to_s.empty?
-        secrets[secret] = ENV[secret.upcase]
-        # we don't want to store Environment variables
-        generate_file = false
-      end
-      secrets[secret] = SecureRandom.hex if secrets[secret].to_s.empty?
-      ENV["TF_VAR_#{secret}"] = secrets[secret]
-    end
+  SECRETS_FILE = "secrets.yaml"
+
+  def self.collect_secrets()
+    ENV['TF_VAR_keyring_name'] = Secrets::KMS_KEYRING
+    $compose_env << "TF_VAR_keyring_name"
 
-    if generate_file
-      puts "Secret file #{saved_secrets_file_path} for this deployment not found. I will create one."
-      File.open(saved_secrets_file_path, 'w+') do |file|
-        file.write(secrets.to_yaml)
+    encryption_keys = []
+    secrets = {}
+    %x{find ../../modules -maxdepth 2 | grep #{Secrets::SECRETS_FILE}}.split.each do |mod|
+      mod_default_key = File.dirname(mod).split('/').last
+      mod_secrets_cfg = YAML.load(File.read(mod))
+
+      kms_key = mod_secrets_cfg['kms_key'] ? mod_secrets_cfg['kms_key'] : mod_default_key
+      encryption_keys << "\"#{kms_key}\"" unless encryption_keys.include? kms_key
```

Comment:
I usually don't mind any quotes style, while code does its job and remains readable. `'"#{kms_key}"'` is not going to work here, because single quote strings are not subject to [string interpolation](https://en.wikibooks.org/wiki/Ruby_Programming/Syntax/Literals#Strings). `%Q{}` notation in this case definitely looks less readable to me than escaped quotes, but we can use it, if you prefer...